executor/index_lookup_join_test.go

```diff
@@ -212,3 +213,41 @@ func (s *testSuite5) TestIssue16887(c *C) {
 	rows = tk.MustQuery("show warnings").Rows()
 	c.Assert(len(rows) > 0, Equals, true)
 }
+
+func (s *testSuite5) TestIndexJoinEnumSetIssue19233(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t;")
+	tk.MustExec("drop table if exists i;")
+	tk.MustExec("drop table if exists p1;")
+	tk.MustExec("drop table if exists p2;")
+	tk.MustExec(`CREATE TABLE p1 (type enum('HOST_PORT') NOT NULL, UNIQUE KEY (type)) ;`)
```

Comment:
would the error occur if enum == set ? or decimal, json some types?

Reply:
Thanks, I found a new bug, https://github.com/pingcap/tidb/issues/19410